core/src/main/java/com/sequenceiq/cloudbreak/rotation/context/SaltRunOrchestratorStateRotationContext.java

```diff
@@ -0,0 +1,195 @@
+package com.sequenceiq.cloudbreak.rotation.context;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import com.sequenceiq.cloudbreak.orchestrator.host.OrchestratorStateParams;
+import com.sequenceiq.cloudbreak.orchestrator.host.OrchestratorStateRetryParams;
+import com.sequenceiq.cloudbreak.orchestrator.model.GatewayConfig;
+import com.sequenceiq.cloudbreak.orchestrator.state.ExitCriteriaModel;
+
+public class SaltRunOrchestratorStateRotationContext extends SaltStateApplyRotationContext {
+
+    private static final int DEFAULT_RETRY = 1;
+
+    private final Map<String, Object> rotateParams;
+
+    private final Optional<Map<String, Object>> rollbackParams;
+
+    private final Optional<Map<String, Object>> cleanupParams;
+
+    private final Boolean changed;
```

Comment:
what does this mean here? I am not sure, so probably someone else can be confused too

Reply:
I've clarified it, but it would be better checking this in the prevalidation flow step (if it will be finished)